py/test/selenium/webdriver/common/virtual_authenticator_tests.py

```diff
@@ -144,109 +152,240 @@ def test_add_and_remove_non_resident_credentials(driver, pages):
 @pytest.mark.xfail_firefox
 @pytest.mark.xfail_safari
 @pytest.mark.xfail_remote
-def test_add_and_remove_resident_credentials(driver, pages):
-    driver = create_rk_enabled_authenticator(driver)
+def test_add_non_resident_credential_when_authenticator_uses_u2f_protocol(driver, pages):
+    driver = create_rk_disabled_u2f_authenticator(driver)
+    driver.get(pages.url("virtual-authenticator.html", localhost=True))
 
-    pages.load("virtual-authenticator.html")
-    assert driver.virtual_authenticator_id is not None
+    base64_pk = '''
+    MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQg8_zMDQDYAxlU-Q
+    hk1Dwkf0v18GZca1DMF3SaJ9HPdmShRANCAASNYX5lyVCOZLzFZzrIKmeZ2jwU
+    RmgsJYxGP__fWN_S-j5sN4tT15XEpN_7QZnt14YvI6uvAgO0uJEboFaZlOEB
+    '''
 
     credential = Credential.create_non_resident_credential(
         bytearray({1, 2, 3, 4}),
         "localhost",
-        b64decode(BASE64__ENCODED_PK),
+        urlsafe_b64decode(base64_pk),
         0,
     )
-
-    credential2 = Credential.create_resident_credential(
-        bytearray({1, 2, 3, 4, 5}),
-        "localhost",
-        bytearray({1}),
-        b64decode(BASE64__ENCODED_PK),
-        1,
-    )
-
     driver.add_credential(credential)
-    assert len(driver.get_credentials()) == 1
-
-    driver.add_credential(credential2)
-    assert len(driver.get_credentials()) == 2
-
-    driver.remove_credential(credential.id)
-    assert len(driver.get_credentials()) == 1
+    assert get_assertion_for(driver, [1, 2, 3, 4]).get('status', '') == 'OK'
 
     driver.remove_virtual_authenticator()
 
 
 @pytest.mark.xfail_firefox
 @pytest.mark.xfail_safari
 @pytest.mark.xfail_remote
-def test_remove_all_credentials(driver):
-    options = VirtualAuthenticatorOptions()
-    options.has_resident_key = True
+def test_add_resident_credential_not_supported_when_authenticator_uses_u2f_protocol(driver, pages):
+    driver = create_rk_enabled_u2f_authenticator(driver)
+    driver.get(pages.url("virtual-authenticator.html", localhost=True))
 
-    driver.add_virtual_authenticator(options)
-    assert driver.virtual_authenticator_id is not None
+    base64_pk = '''
+    MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQg8_zMDQDYAxlU-Q
+    hk1Dwkf0v18GZca1DMF3SaJ9HPdmShRANCAASNYX5lyVCOZLzFZzrIKmeZ2jwU
+    RmgsJYxGP__fWN_S-j5sN4tT15XEpN_7QZnt14YvI6uvAgO0uJEboFaZlOEB
+    '''
 
-    credential = Credential.create_non_resident_credential(
+    credential = Credential.create_resident_credential(
         bytearray({1, 2, 3, 4}),
         "localhost",
-        b64decode(BASE64__ENCODED_PK),
+        bytearray({1}),
+        urlsafe_b64decode(base64_pk),
         0,
     )
+    try:
+        driver.add_credential(credential)
```

Comment:
we should use [`pytest.raises`](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions)